### PR TITLE
Updates for Sitecore XP 10.1

### PR DESCRIPTION
**Pull request by nickwesselman**

### CODE REVIEW

File: custom-images/docker-compose.override.yml [hunk 1]
```diff
@@ -28,16 +28,21 @@ services:
     volumes:
       - ${LOCAL_DATA_PATH}\mssql:c:\data
   
+  # Mount our Solr data folder and use our retagged Solr image.
   solr:
-    image: ${REGISTRY}${COMPOSE_PROJECT_NAME}-xp0-solr:${VERSION:-latest}
+    volumes:
+      - type: bind
+        source: .\docker\data\solr
+        target: c:\data
```
Comment (ambrauer) [inline custom-images/docker-compose.override.yml:36]: Use our LOCAL_DATA_PATH here?
```suggestion
      - ${LOCAL_DATA_PATH}\solr:c:\data
```

File: custom-images/docker-compose.override.yml [hunk 2]
```diff
@@ -28,16 +28,21 @@ services:
     volumes:
       - ${LOCAL_DATA_PATH}\mssql:c:\data
   
+  # Mount our Solr data folder and use our retagged Solr image.
```
Comment (ambrauer) [inline custom-images/docker-compose.override.yml:31]: There is no "retagged Solr image", right? Just mounting data folder.

File: custom-images/docker-compose.xm1.override.yml [hunk 3]
```diff
@@ -35,16 +35,21 @@ services:
     volumes:
       - ${LOCAL_DATA_PATH}\mssql:c:\data
   
+  # Mount our Solr data folder and use our retagged Solr image.
```
Comment (ambrauer) [inline custom-images/docker-compose.xm1.override.yml:38]: Same here (update comment and use LOCAL_DATA_PATH for mount)

File: custom-images/docker-compose.xp1.override.yml [hunk 4]
```diff
@@ -35,16 +35,21 @@ services:
     volumes:
       - ${LOCAL_DATA_PATH}\mssql:c:\data
   
+  # Mount our Solr data folder and use our retagged Solr image.
```
Comment (ambrauer) [inline custom-images/docker-compose.xp1.override.yml:38]: Same here (update comment and use LOCAL_DATA_PATH for mount)

File: custom-images/docker-compose.override.yml [hunk 5]
```diff
@@ -28,16 +28,21 @@ services:
...
+      context: ./docker/build/solr-init
       args:
-        BASE_IMAGE: ${SITECORE_DOCKER_REGISTRY}sitecore-xp0-solr:${SITECORE_VERSION}
+        BASE_IMAGE: ${SITECORE_DOCKER_REGISTRY}sitecore-xp0-solr-init:${SITECORE_VERSION}
         SXA_IMAGE: ${SITECORE_MODULE_REGISTRY}sxa-xp1-assets:${SXA_VERSION}
-    mem_limit: 1GB
-    volumes:
-      - ${LOCAL_DATA_PATH}\solr:c:\data
 
   id:
```
Comment (ambrauer) [inline custom-images/docker-compose.override.yml:47]: Set SITECORE_DEVELOPMENT_PATCHES: CustomErrorsOff for 'id'? Noticed we're doing it for the other topologies.

File: custom-images/init.ps1 [hunk 6]
```diff
@@ -37,14 +37,15 @@ if (-not $SitecoreGallery) {
...
-$dockerToolsVersion = "10.0.5"
+$dockerToolsVersion = "10.1.4"
 Remove-Module SitecoreDockerTools -ErrorAction SilentlyContinue
 if (-not (Get-InstalledModule -Name SitecoreDockerTools -RequiredVersion $dockerToolsVersion -ErrorAction SilentlyContinue)) {
     Write-Host "Installing SitecoreDockerTools..." -ForegroundColor Green
     Install-Module SitecoreDockerTools -RequiredVersion $dockerToolsVersion -Scope CurrentUser -Repository $SitecoreGallery.Name
 }
 Write-Host "Importing SitecoreDockerTools..." -ForegroundColor Green
 Import-Module SitecoreDockerTools -RequiredVersion $dockerToolsVersion
+Write-SitecoreDockerWelcome
```
Comment (ambrauer) [inline custom-images/init.ps1:48]: 👍

**Review by ambrauer — CHANGES_REQUESTED** [review]

File: custom-images/docker-compose.override.yml (hunk 5 above)
Comment (nickwesselman) [reply custom-images/docker-compose.override.yml:47]: hmm should just be on CM and CD

Comment (ambrauer) [reply custom-images/docker-compose.override.yml:47]: Ah yes ignore this, misread the diff :)

File: custom-images/docker-compose.override.yml (hunk 1 above)
Comment (nickwesselman) [reply custom-images/docker-compose.override.yml]: done!

**Review by ambrauer — APPROVED** [review]

**Pull request merged**